app/controllers/components/course/materials_component.rb

```diff
@@ -0,0 +1,15 @@
+class Course::MaterialsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def sidebar_items
+    [
+      {
+        key: :materials,
+        title: I18n.t('course.material.sidebar_title'),
+        weight: 4,
+        path: course_material_folder_path(current_course, current_course.root_folder),
```

Comment:
This is not going to be a problem that will not recur. You will have changes to the schema and data even after deployment. No point trying to run away from it.

I already said you cannot use models.